DataTransformation/AggregateTransformation.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Betlln.Data.Integration.Core;
+
+namespace Betlln.Data.Integration
+{
+    public class AggregateTransformation : DataFeed
+    {
+        //TODO: ensure column not added twice
+        private readonly List<DataElementPairing> _columns;
+
+        internal AggregateTransformation()
+        {
+            _columns = new List<DataElementPairing>();
+        }
+
+        public DataFeed DataSource { get; set; }
+
+        public void AddGroupingColumn(string sourceColumnName, string outputColumnName = null)
+        {
+            _columns.Add(new DataElementPairing(sourceColumnName, outputColumnName ?? sourceColumnName, AggregationRole.GroupingKey));
+        }
+
+        public void AddSumColumn(string sourceColumnName, string outputColumnName = null)
+        {
+            _columns.Add(new DataElementPairing(sourceColumnName, outputColumnName ?? sourceColumnName, AggregationRole.Sum));
+        }
+
+        protected override IDataRecordIterator CreateReader()
+        {
+            List<DataRecord> aggregatedRecords = new List<DataRecord>();
+            
+            List<DataElementPairing> dimensionColumns = _columns.Where(columnInfo => columnInfo.AggregationRole == AggregationRole.GroupingKey).ToList();
+            List<DataElementPairing> factColumns = _columns.Where(columnInfo => columnInfo.AggregationRole != AggregationRole.GroupingKey).ToList();
+
+            using (IDataRecordIterator source = DataSource.GetReader())
+            {
+                foreach(DataRecord sourceRecord in source)
+                {
+                    DataRecord dataRecord = aggregatedRecords.Find(resultRecord => 
+                    {
+                        foreach (DataElementPairing dimensionColumn in dimensionColumns)
+                        {
+                            object expectedValue = resultRecord[dimensionColumn.DestinationName];
+                            object actualValue = sourceRecord[dimensionColumn.SourceName];
+                            if (!Equals(expectedValue, actualValue))
+                            {
+                                return false;
```

Comment:
suggestion toss the return to the end of the function. Its growing and this will likely cause a bug. Seen it many times before